fairmq/shmem/Message.h

```diff
@@ -266,8 +266,12 @@ class Message final : public fair::mq::Message
             if (fMeta.fShared < 0) { // UR msg is not yet shared
                 return 1;
             } else {
-                fManager.GetSegment(fMeta.fSegmentId);
-                return ShmHeader::RefCount(fManager.GetAddressFromHandle(fMeta.fShared, fMeta.fSegmentId));
+                fRegionPtr = fManager.GetRegionFromCache(fMeta.fRegionId);
+                if (fRegionPtr) {
+                    return fRegionPtr->GetRefCount(fRegionPtr->GetRefCountAddressFromHandle(fMeta.fShared));
+                } else {
+                    throw TransportError(tools::ToString("Cannot get unmanaged region with id ", fMeta.fRegionId));
+                }
```

Comment:
```suggestion
            if (fMeta.fShared < 0) { // UR msg is not yet shared
                return 1;
            }
            fRegionPtr = fManager.GetRegionFromCache(fMeta.fRegionId);
            if (!fRegionPtr) {
                throw TransportError(tools::ToString("Cannot get unmanaged region with id ", fMeta.fRegionId));
            }
            return fRegionPtr->GetRefCount(fRegionPtr->GetRefCountAddressFromHandle(fMeta.fShared));
```
`readability-else-after-return`